solr/contrib/ltr/src/java/org/apache/solr/ltr/log/FeatureLogger.java

```diff
@@ -112,16 +111,15 @@ public FV_TYPE getFeatureVector(int docid, ModelQuery reRankModel,
   public static class MapFeatureLogger extends FeatureLogger<Map<String,Float>> {
 
     @Override
-    public Map<String,Float> makeFeatureVector(String[] featureNames,
-        float[] featureValues, boolean[] featuresUsed) {
+    public Map<String,Float> makeFeatureVector(LinkedHashMap<Integer, FeatureInfo> featuresInfo) {
       Map<String,Float> hashmap = Collections.emptyMap();
-      if (featureNames.length > 0) {
-        hashmap = new HashMap<String,Float>(featureValues.length);
-        for (int i = 0; i < featuresUsed.length; i++) {
-          if (featuresUsed[i]) {
-            hashmap.put(featureNames[i], featureValues[i]);
-          }
-        }
+      if (featuresInfo.size() > 0) {
+         hashmap = new HashMap<String,Float>(featuresInfo.size());
+         for (FeatureInfo featInfo:featuresInfo.values()){  
+            if (featInfo.getUsed()) {
```

Comment:
If we make the map contain only used features, then we dont need these checks.  The memory footprint will also be smaller if you have a feature space where only a small subset gets triggered for any given document.